### PR TITLE
report accurate mean instead of an approximation

### PR DESCRIPTION
**Pull request by Dieterbe**

### CODE REVIEW

**Comment by woodsaj** [comment]
I dont think this will work.  Your sum will quickly overflow and you won't get meaning values anymore.

**Comment by woodsaj** [comment]
Actually as you are using microseconds,  this will take a while to overflow, but it will still eventually overflow.

**Comment by shanson7** [comment]
A 64-bit uint has a enormous max value. Even if every microsecond of every day was counted it would take ~3.5 million years to overflow. Even if this were updated from 1000 threads all day every day, it would still not overflow before the hardware rotted :).

**Comment by Dieterbe** [comment]
FTR math.MaxUint64 is 18446744073709551615. should be plenty to track the sum for each report interval (between 10~60s)
i just saw one tweak I can do:
```
-		sum := atomic.LoadUint64(&l.sum)
-		atomic.StoreUint64(&l.sum, 0)
+		sum := atomic.SwapUint64(&l.sum, 0)
```
will rebase and merge. thanks guys

File: stats/latencyhistogram15s32.go [hunk 1]
```diff
@@ -30,8 +33,9 @@ func (l *LatencyHistogram15s32) ReportGraphite(prefix, buf []byte, now time.Time
 	// for now, only report the summaries :(
 	r, ok := l.hist.Report(snap)
 	if ok {
+		sum := atomic.SwapUint64(&l.sum, 0)
```
Comment (woodsaj) [inline stats/latencyhistogram15s32.go]: you are reseting sum to 0 on every flush, but not count.

Comment (Dieterbe) [reply stats/latencyhistogram15s32.go]: r.Count comes from the report which comes from the histogram snapshot. the histogram is reset when snapshot is taken

**Pull request merged**